File: skill_sdk/tests/fixtures/java-project/src/main/java/com/fixture/shop/models/OrderItem.java
package com.fixture.shop.models;

/**
 * A single line item within an order. Mirrors OrderItem in shop/models.go.
 */
public class OrderItem {

    private final Product product;
    private final int quantity;
    private final double unitPrice;

    public OrderItem(Product product, int quantity, double unitPrice) {
        this.product = product;
        this.quantity = quantity;
        this.unitPrice = unitPrice;
    }

    public Product getProduct() { return product; }

    public int getQuantity() { return quantity; }

    public double getUnitPrice() { return unitPrice; }

    public double subtotal() {
        return unitPrice * quantity;
    }
}

File: skill_sdk/tests/fixtures/js-project/src/shop/models/OrderItem.js
/**
 * A single line item within an order. Mirrors OrderItem in the Java fixture.
 */
export class OrderItem {
  /**
   * @param {import('./Product.js').Product} product
   * @param {number} quantity
   * @param {number} unitPrice
   */
  constructor(product, quantity, unitPrice) {
    /** @type {import('./Product.js').Product} */
    this._product = product;
    /** @type {number} */
    this._quantity = quantity;
    /** @type {number} */
    this._unitPrice = unitPrice;
  }

  /** @returns {import('./Product.js').Product} */
  getProduct() { return this._product; }

  /** @returns {number} */
  getQuantity() { return this._quantity; }

  /** @returns {number} */
  getUnitPrice() { return this._unitPrice; }

  /** @returns {number} */
  subtotal() {
    return this._unitPrice * this._quantity;
  }
}
